### PR TITLE
test the functionality of the components and calculator and operate file

### PR DESCRIPTION
**Pull request by NickEmma**
- I created unit tests for all React components:
- I used Jest snapshots to test the components.
- I used React Testing Library to simulate user interaction.
- I ran the full tests suite using Jest. All tests should pass.

### CODE REVIEW

File: src/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import calculate from './calculate';
+
+describe('calculate', () => {
+  it('should return an object', () => {
+    const result = calculate({ total: 0, next: null, operation: null }, 'AC');
+    expect(result).toEqual({ total: 0, next: null, operation: null });
```
Comment (Ismailco) [inline src/logic/calculate.test.js:6]: - [ ] [OPTIONAL] You can cover more scenarios, on your tests so your function is truly tested and functions as expected in all cases

**Review by Ismailco — APPROVED** [review]
### Hi @NickEmma, @Roseokpe, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally
✔️ Tests passes 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/calculate.test.js (hunk 1 above)
Comment (NickEmma) [reply src/logic/calculate.test.js:6]: okay thanks @Ismailco I will do just that

**Comment by NickEmma** [comment]
@Ismailco Thank you for reviewing and approving our code.

**Pull request merged**